### PR TITLE
[DESK-834] open vnc in windows

### PR DESCRIPTION
**Pull request by mauricio0000**
### Description

Launch VNC  in Windows

### Checklist

- [ ] Pull Request title is in the format `[PROJ-12345] Some useful description here...`
- [ ] I have prefixed this PR with `[WIP]` if it is a Work In Progress (not ready to merge)
- [ ] The git branch is in the format `PROJ-12345-some-useful-description-here`
- [ ] I have added one or more reviewers
- [ ] I have added/updated tests for any code changes
- [ ] I have updated documentation (including README, inline comments and docs.remote.it docs)
- [ ] This PR only contains one isolated change (bug fix, feature, etc)
- [ ] Translation strings added/updated for all user-visible text
- [ ] I have manually reviewed this PR to make sure only the files I intended to change were changed and nothing else
- [ ] I have notified reviewers on Slack by @ mentioning them in the `#desktop`channel
- [ ] I have moved the Jira ticket into `Resolved` state and made a note in the ticket with a link to this PR

### Test plan

1. Sign in in Windows environment with VNC installed
2. Go to devices
3. Choose a device with VNC service
4. Connect and open VNC service
5. Launch service
4. You should see VNC open

### Ticket

https://remoteit.atlassian.net/browse/DESK-834

### CODE REVIEW

File: frontend/src/components/DialogApp/DialogApp.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  const getApp = () => {
+    window.open(link)
+    closeAll()
+  }
+  return (
+    <>
+      <Dialog open={openApp} onClose={closeAll} maxWidth="xs" fullWidth>
+        <Typography variant="h1">
+          Please install {App} to launch {type} connections.
+        </Typography>
```
Comment (JamieRuderman) [inline frontend/src/components/DialogApp/DialogApp.tsx:20]: I think this should be in a <DialogTitle> element.. .please refer to the PromptModal component. Thank you!

File: backend/src/launch.ts [hunk 2]
```diff
@@ -1,54 +1,92 @@
...
+      EventBus.emit(EVENTS.notInstalledPutty, { install: true, loading: false })
+    }
+  })
+}
+
+export const openCMDvnc = (launchApp: any) => {
+  if (launchApp.pathVNC) return launchVNC(launchApp.pathVNC, launchApp.port, launchApp.host, launchApp.username)
+  EventBus.emit(EVENTS.notInstalledVNC, { install: false, loading: true })
+  Logger.info('LAUNCH APP', { launchApp })
+  const process = child_process.exec(`DIR /S vncviewer.exe /B`, { cwd: 'C:\\' })
```
Comment (JamieRuderman) [inline backend/src/launch.ts:49]: This search performed very slowly for me - if there is a faster way that would be great!

File: backend/src/launch.ts [hunk 3]
```diff
@@ -1,54 +1,92 @@
...
+    Logger.info(`child process exited with code ${code}`)
+  })
+
+  child_process.exec(`DIR /S vncviewer.exe /B`, { cwd: 'C:\\' }, (error: any, stdout: any, stderr: any) => {
+    error && Logger.error(`error: ${error}`)
+    Logger.info(`RESULT VNC VIEWER`, { stdout, stderr })
+    if (!stdout.trim()) {
+      EventBus.emit(EVENTS.notInstalledVNC, { install: true, loading: false })
+    }
+  })
```
Comment (JamieRuderman) [inline backend/src/launch.ts:73]: This code looks very similar to the putty launch code, if we can combine and remove the duplication that would be great. We need as general as possible way to launch or run commands Windows.

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
Please see notes thanks!

File: backend/src/launch.ts (hunk 2 above)
Comment (mauricio0000) [reply backend/src/launch.ts]: we had the same issue in SSH, I remember that we were talking and we didn't find any improve there

File: backend/src/sharedCopy/applications.ts [hunk 4]
```diff
@@ -120,7 +120,7 @@ function getApplication(typeID?: number) {
       app = new Application({
         title: 'VNC',
         icon: 'desktop',
-        defaultLaunchTemplate: 'vnc://[host]:[port]',
+        defaultLaunchTemplate: '[username] [host]:[port]',
```
[the remark this answers is not in the document]
Comment (mauricio0000) [reply backend/src/sharedCopy/applications.ts]: its working

**Review by JamieRuderman — APPROVED** [review]

**Pull request merged**